internal/controllers/objecttemplate/template_reconciler.go

```diff
@@ -101,15 +105,18 @@ func (r *templateReconciler) Reconcile(
 		return res, fmt.Errorf("updating status conditions from owned object: %w", err)
 	}
 
-	obj.SetOwnerReferences(existingObj.GetOwnerReferences())
-	obj.SetLabels(labels.Merge(existingObj.GetLabels(), obj.GetLabels()))
-	obj.SetAnnotations(labels.Merge(existingObj.GetAnnotations(), obj.GetAnnotations()))
-
-	obj.SetResourceVersion(existingObj.GetResourceVersion())
-	if err := r.client.Update(ctx, obj); err != nil {
-		return res, fmt.Errorf("updating templated object: %w", err)
+	ownedByPKO := utils.HasSameController(existingObj, obj)
+	specChanged := !equality.Semantic.DeepEqual(existingObj.Object["spec"], obj.Object["spec"])
```

Comment:
That's correct @ashishmax31

Reply:
So the equality check would need to be done on all top level keys of the object except `metadata` and `status` instead of just on `spec`.

Reply:
Yes, something like this could work @venkateshsredhat 
```
func shouldUpdate(existingObj, desiredObject *unstructured.Unstructured) bool {
	for field, currentValue := range desiredObject.Object {
		if field == "metadata" || field == "status" {
			continue
		}
		existingValue := existingObj.Object[field]
		if !equality.Semantic.DeepEqual(existingValue, currentValue) {
			return true
		}
	}
	return false
}
```